src/lu.cpp

```diff
@@ -180,6 +180,20 @@ JNIEXPORT void JNICALL Java_com_powsybl_math_matrix_SparseLUDecomposition_update
         if (ok == 0) {
             throw std::runtime_error("klu_refactor error " + context.error());
         }
+        ok = klu_rgrowth(ap.get(), ai.get(), ax.get(), context.symbolic, context.numeric, &context.common);
+        if (ok == 0) {
+            throw std::runtime_error("klu_rgrowth error " + context.error());
+        }
+        // if rgrowth is too small we have to do a whole factorization
+        if (context.common.rgrowth < 1e-10) { // no idea what could be the right threshold but 1e-10 seems to work...
```

Comment:
The documentation says:
> Computes the reciprocal pivot growth, `rgrowth = min j ((max i |c ij |)/(max i |u ij |))`, where `c ij` is a
scaled entry in a diagonal block of the block triangular form. In MATLAB notation:
`rgrowth = min (max (abs (R\A(p,q) - F)) ./ max (abs (U)))`
where the factorization is `L*U + F = R \ A(p,q).`